### PR TITLE
Added support for user agent and platform

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

### CODE REVIEW

File: lib/Formatter.php [hunk 1]
```diff
@@ -21,14 +24,32 @@
...
+		'IE' => 				Request::USER_AGENT_IE,
+		'Edge' => 				Request::USER_AGENT_MS_EDGE,
+		'Chrome' => 			Request::USER_AGENT_CHROME,
+		'Firefox' => 			Request::USER_AGENT_FIREFOX,
+		'Safari' => 			Request::USER_AGENT_SAFARI,
+		'Chrome for Android' => Request::USER_AGENT_ANDROID_MOBILE_CHROME,
+		'Android' => 			IRequest::USER_AGENT_CLIENT_ANDROID,
+		'Talk for Android' =>	IRequest::USER_AGENT_TALK_ANDROID,
+		'Desktop client' => 	IRequest::USER_AGENT_CLIENT_DESKTOP,
+		'IOS client' => 		IRequest::USER_AGENT_CLIENT_IOS,
```
Comment (MorrisJobke) [inline lib/Formatter.php]: It's written `iOS`:see_no_evil:

File: lib/Formatter.php [hunk 2]
```diff
@@ -21,14 +24,32 @@
...
 
 	protected $timestamp;
 
+	protected $userAgents = [
+		'IE' => 				Request::USER_AGENT_IE,
+		'Edge' => 				Request::USER_AGENT_MS_EDGE,
+		'Chrome' => 			Request::USER_AGENT_CHROME,
+		'Firefox' => 			Request::USER_AGENT_FIREFOX,
+		'Safari' => 			Request::USER_AGENT_SAFARI,
+		'Chrome for Android' => Request::USER_AGENT_ANDROID_MOBILE_CHROME,
```
Comment (blizzz) [inline lib/Formatter.php]: ^ we should look into moving those to IRequest, too, and perhaps also offer an array, maybe with additional information as type ("browser", "desktop", "other-app"? at least DAV app needs browsers). That way, changes only stay in one place and we don't need to use private API in this app.

File: lib/Formatter.php [hunk 3]
```diff
@@ -75,4 +97,16 @@ public function humanFileSize($bytes) {
 		return "$bytes PB";
 	}
 
+	public function getPlatformFromUA(string $userAgent) {
+		if ($userAgent === '') {
+			return 'None';
+		}
+		foreach ($this->userAgents as $platform => $regex) {
+			if (preg_match($regex, $userAgent)) {
+				return $platform;
```
Comment (blizzz) [inline lib/Formatter.php]: platform cannot be translated this way

Comment (skjnldsv) [reply lib/Formatter.php]: Should it though? I guess the occ output is supposed to be used for an outside api or tool?

Comment (blizzz) [reply lib/Formatter.php]: ok, i thought it would be presented to the end user as is. then, never mind

**Comment by MorrisJobke** [comment]
Works, but it only lists the last active platform. Wouldn't it be better to show all available platforms somehow? Like `Safari|Firefox`?

File: lib/Formatter.php [hunk 4]
```diff
@@ -21,14 +24,32 @@
...
+		'Firefox' => 			Request::USER_AGENT_FIREFOX,
+		'Safari' => 			Request::USER_AGENT_SAFARI,
+		'Chrome for Android' => Request::USER_AGENT_ANDROID_MOBILE_CHROME,
+		'Android' => 			IRequest::USER_AGENT_CLIENT_ANDROID,
+		'Talk for Android' =>	IRequest::USER_AGENT_TALK_ANDROID,
+		'Desktop client' => 	IRequest::USER_AGENT_CLIENT_DESKTOP,
+		'iOS client' => 		IRequest::USER_AGENT_CLIENT_IOS,
+		'Talk for iOS' => 		IRequest::USER_AGENT_TALK_IOS,
+		'Outlook' => 			IRequest::USER_AGENT_OUTLOOK_ADDON,
+		'Thunderbird' => 		IRequest::USER_AGENT_THUNDERBIRD_ADDON
```
Comment (nickvergessen) [inline lib/Formatter.php]: These do not exist in 12, im not sure if we backported them to 13.

**Comment by skjnldsv** [comment]
@blizzz should we merge?

**Comment by nickvergessen** [comment]
No we shouldn't

**Pull request closed without merging**